lib/src/db/model_db_impl.dart

```diff
@@ -238,6 +238,17 @@ class ModelDBImpl implements ModelDB {
     }
   }
 
+  /// Returns true if a  constructor invocation is valid even if the specified
+  /// [parameter] is omitted.
+  ///
+  /// This is  true for named parameters, optional parameters, and parameters
+  /// with a default value.
+  static bool _canBeOmitted(mirrors.ParameterMirror parameter) {
+    return parameter.isOptional ||
+        parameter.isNamed ||
+        parameter.hasDefaultValue;
```

Comment:
I'm not sure if it makes sense to check this, but does it make a difference if the parameter is `@required`?  Does mirrors even give you access to that?

Reply:
Done (and test added).